colcon_core/task/python/build.py

```diff
@@ -119,10 +119,7 @@ def __init__(self):  # noqa: D107
                 '--single-version-externally-managed',
             ]
             self._append_install_layout(args, cmd)
-            cmd += [
-                'bdist_egg', '--dist-dir', os.path.join(
-                    args.build_base, 'dist'),
-            ]
```

Comment:
> As it turns out, bdist_egg is only necessary to not pollute the source path when *not* doing --single-value-externally-managed.

Can you clarify this? The code doesn't explicitly pass `--single-value-externally-managed` anywhere - is that happening implicitly? Otherwise I don't understand the rational why it isn't necessary if we don't use the option.

Reply:
The line above. It's being passed to the `install` command. The install command calls `bdist_egg`, depending on whether or not the --single-version-externally-managed flag is passed.